src/store-update.common.ts

```diff
@@ -104,21 +108,30 @@ export class StoreUpdateCommon {
 
   buildDialogOptions({ skippable = true } = {}): ConfirmOptions {
     let options = {
-      message: LocalesHelper.translate('ALERT_MESSAGE'),
+      message: this.getMessage(this._alertOptions.message, LocalesHelper.translate('ALERT_MESSAGE')),
       neutralButtonText: null,
-      okButtonText: LocalesHelper.translate('ALERT_UPDATE_BUTTON'),
-      title: LocalesHelper.translate('ALERT_TITLE'),
+      okButtonText: this.getMessage(this._alertOptions.updateButton, LocalesHelper.translate('ALERT_UPDATE_BUTTON')),
+      title: this.getMessage(this._alertOptions.title, LocalesHelper.translate('ALERT_TITLE')),
     }
 
     if (skippable) {
       options = {
         ...options,
-        neutralButtonText: LocalesHelper.translate('ALERT_SKIP_BUTTON'),
+        neutralButtonText: this.getMessage(this._alertOptions.skipButton, LocalesHelper.translate('ALERT_SKIP_BUTTON')),
       }
     }
+
     return options
   }
 
+  getMessage(text: string, fallback: string): string {
+    if (this._alertOptions && this._alertOptions.custom) {
+      return (text == "" || !text) ? fallback : text; //If the custom alert text is not set, return the fallback
+    } else {
+      return fallback;
+    }
+  }
```

Comment:
I don't get the point of setting a `custom` boolean entry that defines if either we overrides the locales or not. 
If there's an `alertOptions.title` defined, then it should be enough to determine what string you should use.

Reply:
Quickly, with my remarks above, I'd tend to use something more this way =>

```
this._getMessage('title', 'ALERT_TITLE')

private _getMessage(alertOptionEntry: string, fallbackTranslateKey: string): string {
  if this._hasValidAlertOptionEntry(alertOptionEntry) return this._alertOptions[key]
  return LocalesHelper.translate(fallbackTranslateKey)
}

private _hasValidAlertOptionEntry(key) {
  if (!this._alertOptions) return false
  return this._alertOptions[key] !== '' || this._alertOptions[key]
}
```

This is a quick draft. what do you think of this?